### ISSUE TITLE
[localizations] Bug Placeholder type int need format if other placeholder type is double or num

### ISSUE THREAD

**Issue by lsaudon** [flutter/flutter#115989]
## Description

Why the placeholder with an int type does not need a format when it is alone, but needs a format when there are other placeholders.

## Steps to Reproduce

1. Execute `flutter create my_app` on the code sample 
2. Add `flutter_localizations` and `intl` and `generate: true`:
  ```yml
name: my_app
description: My App
version: 1.0.0+1
publish_to: none

environment:
  sdk: '>=2.18.2 <3.0.0'
  flutter: 3.3.6

dependencies:
  flutter:
    sdk: flutter
  flutter_localizations:
    sdk: flutter
  intl: ^0.17.0

flutter:
  uses-material-design: true
  generate: true

  ```

3. Add l10n.yaml file:
  ```yml
arb-dir: lib/l10n/arb
template-arb-file: app_en.arb
output-localization-file: app_localizations.dart
nullable-getter: false
  ```

4. Add `lib\l10n\arb\app_en.arb`
```json
{
  "@@locale": "en",
  "text": "a {number} {otherNumber}",
  "@text": {
    "placeholders": {
      "number": {
        "type": "int"
      },
      "otherNumber": {
        "type": "num",
        "format": "compact"
      }
    }
  }
}
```

**Expected results:** 

No error

**Actual results:** 
```sh
Generating synthetic localizations package failed with 1 error:

Exception: Number format null for the number placeholder does not have a corresponding NumberFormat constructor.
Check the intl library's NumberFormat class constructors for allowed number formats.
```

**Comment by huycozy** [comment]
@lsaudon As I see, there is missing `format` property for `number` placeholder, you can try this one:

```arb
"text": "a {number} {otherNumber}",
"@text": {
  "placeholders": {
    "number": {
      "type": "int",
      "format": "compact"
    },
    "otherNumber": {
      "type": "num",
      "format": "compact"
    }
  }
}
```

**Comment by lsaudon** [comment]
@huycozy I know this works, but compact is not the same as no format.

Why the code below works and not with `num` or `double` type in other placeholders ?

```json
"text": "a {number} {otherNumber}",
"@text": {
  "placeholders": {
    "number": {
      "type": "int"
    },
    "otherNumber": {
      "type": "string"
    }
  }
}
```

or

```json
"text": "a {number}",
"@text": {
  "placeholders": {
    "number": {
      "type": "int"
    }
  }
}
```

**Comment by lsaudon** [comment]
I would like to have in a translation, an integer without format and a double for a price like:

```json
{
  "@@locale": "en",
  "customerNumberPaid": "Customer number {number} paid {price}",
  "@customerNumberPaid": {
    "placeholders": {
      "number": {
        "type": "int"
      },
      "price": {
        "type": "double",
        "format": "currency"
      }
    }
  }
}
```

Here is the difference between no format and compact.
![image](https://user-images.githubusercontent.com/25029876/203987966-379b1596-5b3e-41e9-bae3-619e9057dc11.png)

```json
{
  "@@locale": "en",
  "customerNumberPaid": "Customer number {number} paid {price}",
  "@customerNumberPaid": {
    "placeholders": {
      "number": {
        "type": "int",
        "format": "compact"
      },
      "price": {
        "type": "double",
        "format": "currency"
      }
    }
  },
  "customerNumber": "Customer number {number}",
  "@customerNumber": {
    "placeholders": {
      "number": {
        "type": "int"
      }
    }
  }
}
```

I don't understand why you put a `int` without format when number is alone, but if there is a `double` it is not possible.

**Comment by huycozy** [comment]
@lsaudon thanks for providing more info, I see the issue now. It's reproducible on Flutter stable `3.3.9` but it seems to be fixed on Flutter master channel `3.6.0`. Please retry and confirm the issue has been fixed. 

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.3.9, on macOS 13.0 22A380 darwin-x64, locale en-VN)
    • Flutter version 3.3.9 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b8f7f1f986 (31 hours ago), 2022-11-23 06:43:51 +0900
    • Engine revision 8f2221fbef
    • Dart version 2.18.5
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Community Edition (version 2022.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 70.0.5
    • Dart plugin version 222.4167.21

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.73.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.52.0

[✓] Connected device (3 available)
    • SM T225 (mobile) • R9JT3004VRJ • android-arm64  • Android 12 (API 31)
    • macOS (desktop)  • macos       • darwin-x64     • macOS 13.0 22A380 darwin-x64
    • Chrome (web)     • chrome      • web-javascript • Google Chrome 107.0.5304.110

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

```bash
[!] Flutter (Channel master, 3.6.0-7.0.pre.52, on macOS 13.0 22A380 darwin-x64, locale en-VN)
    • Flutter version 3.6.0-7.0.pre.52 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ff59250dbe (4 hours ago), 2022-11-24 18:08:30 -0500
    • Engine revision 7665ae5184
    • Dart version 2.19.0 (build 2.19.0-429.0.dev)
    • DevTools version 2.19.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Community Edition (version 2022.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 70.0.5
    • Dart plugin version 222.4167.21

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.73.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.52.0

[✓] Connected device (3 available)
    • Pixel 3a (mobile) • 964AY0WL20 • android-arm64  • Android 12 (API 32)
    • macOS (desktop)   • macos      • darwin-x64     • macOS 13.0 22A380 darwin-x64
    • Chrome (web)      • chrome     • web-javascript • Google Chrome 107.0.5304.110

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

```
</details>

**Comment by lsaudon** [comment]
The issue has been fixed on master.
